### PR TITLE
Support icrc1 format in convert-id

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

It can be useful to create or parse addresses in the [ICRC1 format](https://internetcomputer.org/docs/current/references/icrc1-standard#textual-encoding-of-accounts).

# Changes

Add `icrc1` as a supported input and output format to `scripts/convert-id`.

# Tests

1. Unit test script extended.
2. Unit test script added to CI.

# Todos

- [ ] Add entry to changelog (if necessary).
not necessary

### CODE REVIEW

File: scripts/convert-id [hunk 1]
```diff
@@ -116,6 +127,27 @@ function account_identifier_to_hex() {
...
+function icrc1_to_hex() {
+  sed 's@-[^-]*$@@' | text_to_hex
+}
+
+function hex_to_icrc1() {
+  hex=$(cat)
+
+  subaccount_hex="$(get_subaccount_hex)"
+
+  subaccount_hex="$(echo -n "$subaccount_hex" | sed -e 's@^0*@@')"
```
Comment (mstrasinskis) [inline scripts/convert-id:139]: Is it possible to use the same approach as on line #92?

**Review by mstrasinskis — APPROVED** [review]
Rubber stamp.

File: scripts/convert-id (hunk 1 above)
Comment (dskloetd) [reply scripts/convert-id:145]: This removes leading zeros.
Line 92 adds leading zeros.
I'll add a comment.

Comment (mstrasinskis) [reply scripts/convert-id:145]: True, thanks.

**Pull request merged**